generative/maisi/scripts/infer_controlnet.py

```diff
@@ -0,0 +1,207 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import logging
+import os
+import sys
+from datetime import datetime
+from pathlib import Path
+
+import torch
+import torch.distributed as dist
+import torch.nn.functional as F
+from monai.data import decollate_batch, MetaTensor
+from monai.networks.utils import copy_model_state
+from monai.transforms import SaveImage
+from monai.utils import RankFilter
+
+from .sample import ldm_conditional_sample_one_image
+from .utils import define_instance, load_autoencoder_ckpt, prepare_maisi_controlnet_json_dataloader, setup_ddp
+
+
+def main():
```

Comment:
We can also consider using `@torch.inference_mode()` to simply some settings such as `with torch.no_grad()`

https://pytorch.org/docs/stable/generated/torch.autograd.grad_mode.inference_mode.html

Reply:
Thanks for the suggestion! I add this context-manager to main func.